src/sys/unix/epoll.rs

```diff
@@ -139,12 +139,20 @@ pub mod event {
         (event.events as libc::c_int & libc::EPOLLERR) != 0
     }
 
-    pub fn is_hup(event: &Event) -> bool {
-        (event.events as libc::c_int & libc::EPOLLHUP) != 0
+    pub fn is_read_closed(event: &Event) -> bool {
+        // Both halves of the socket have closed
+        event.events as libc::c_int & libc::EPOLLHUP != 0
+            // Socket has received FIN or called shutdown(SHUT_RD)
+            || (event.events as libc::c_int & libc::EPOLLIN != 0
```

Comment:
Out of curiosity, is it possible to receive `EPOLLRDHUP` w/o receiving `EPOLLIN`?